uaclient/cli/enable.py

```diff
@@ -38,31 +152,76 @@ def action_enable(args, *, cfg, **kwargs):
             option1="--access-only", option2="--variant"
         )
 
-    event.info(messages.REFRESH_CONTRACT_ENABLE)
+    print_fn(messages.REFRESH_CONTRACT_ENABLE)
     try:
         contract.refresh(cfg)
     except (exceptions.ConnectivityError, exceptions.UbuntuProError):
         # Inability to refresh is not a critical issue during enable
         LOG.warning("Failed to refresh contract", exc_info=True)
-        event.warning(warning_msg=messages.E_REFRESH_CONTRACT_FAILURE)
+        warnings.append(
+            {
+                "type": "system",
+                "message": messages.E_REFRESH_CONTRACT_FAILURE.msg,
+                "message_code": messages.E_REFRESH_CONTRACT_FAILURE.name,
+            }
+        )
 
     names = getattr(args, "service", [])
     (
         entitlements_found,
         entitlements_not_found,
     ) = entitlements.get_valid_entitlement_names(names, cfg)
+    enabled_service_names = [
+        s.name for s in _enabled_services(cfg).enabled_services
+    ]
+    all_dependencies = _dependencies(cfg).services
+
     ret = True
-    for ent_name in entitlements_found:
+    for ent_name in entitlements.order_entitlements_for_enabling(
+        cfg, entitlements_found
+    ):
+        real_name = entitlements.get_real_name(cfg, ent_name)
```

Comment:
I think we can create the entitlement object early through the `entitlement_factory` function, which already respects the `valid_names` logic. Then we would need to create it here passing the `called_name` argument to entitlement class. If we do that, then we can fetch the real entitlement name through the `presentation_name` property.

This would allow us to get rid of the `get_real_name` function

Reply:
Coming back to rewrite `get_real_name` and its actually so simple that I've come around to thinking its not necessary.

I'm still going to keep it separate from the later entitlement instantiation, because that will get removed in the next PR anyway.

Reply:
Updated!